### PR TITLE
switch integration tests to external kubeclient

### PR DESCRIPTION
**Pull request by deads2k**

### CODE REVIEW

File: test/integration/deploy_trigger_test.go [hunk 1]
```diff
@@ -689,21 +689,3 @@ func assertEnvVarEquals(name string, value string, deployment *kapi.ReplicationC
 
 	t.Fatalf("Expected env var with name %s and value %s", name, value)
 }
-
-func makeStream(name, tag, dir, image string) *imageapi.ImageStream {
```
Comment (soltysh) [inline test/integration/deploy_trigger_test.go]: Die :smiling_imp:

File: test/integration/node_authorizer_test.go [hunk 2]
```diff
@@ -64,158 +64,161 @@ func TestNodeAuthorizer(t *testing.T) {
...
-	wait.Poll(time.Second, time.Minute, func() (bool, error) {
-		sa, err := superuserClient.Core().ServiceAccounts("ns").Get("default", metav1.GetOptions{})
+	err = wait.Poll(time.Second, time.Minute, func() (bool, error) {
+		sa, err := superuserClient.CoreV1().ServiceAccounts("ns").Get("default", metav1.GetOptions{})
 		if err != nil || len(sa.Secrets) == 0 {
 			return false, nil
 		}
 		return true, nil
 	})
+	if err != nil {
```
Comment (soltysh) [inline test/integration/node_authorizer_test.go]: Good catch :+1:

File: test/integration/oauth_serviceaccount_client_events_test.go [hunk 3]
```diff
@@ -288,26 +286,17 @@ func setupTestSA(client kclientset.Interface, annotationPrefix, annotation strin
...
 	// retry this a couple times.  We seem to be flaking on update conflicts and missing secrets all together
 	err := wait.PollImmediate(30*time.Millisecond, 10*time.Second, func() (done bool, err error) {
-		allSecrets, err := client.Core().Secrets(projectName).List(metav1.ListOptions{})
+		allSecrets, err := client.CoreV1().Secrets(projectName).List(metav1.ListOptions{})
 		if err != nil {
 			return false, err
 		}
 		for i := range allSecrets.Items {
 			secret := &allSecrets.Items[i]
-			secretv1 := &corev1.Secret{}
```
Comment (soltysh) [inline test/integration/oauth_serviceaccount_client_events_test.go]: Yesssss :smiling_imp:

File: vendor/k8s.io/kubernetes/pkg/quota/v1/resources.go [hunk 4]
```diff
@@ -0,0 +1,244 @@
+/*
```
Comment (soltysh) [inline vendor/k8s.io/kubernetes/pkg/quota/v1/resources.go]: I don't see any reasons we would not want to have that on our end and not an upstream, but I might be missing some greater picture you have.

Comment (deads2k) [reply vendor/k8s.io/kubernetes/pkg/quota/v1/resources.go]: > I don't see any reasons we would not want to have that on our end and not an upstream, but I might be missing some greater picture you have.

it comes in with the rebase.  This just makes workable early

Comment (soltysh) [reply vendor/k8s.io/kubernetes/pkg/quota/v1/resources.go]: Can you add that to the list of followups so we don't forget about it?

**Pull request merged**